core/common/src/main/scala/org/typelevel/otel4s/Context.scala

```diff
@@ -0,0 +1,31 @@
+package org.typelevel.otel4s
+
+import cats.Functor
+import cats.effect.kernel.Unique
+import org.typelevel.vault
+
+sealed trait Context {
```

Comment:
This is easy enough, but we need an isomorphism with the Java one to wrap their propagators, and that's where it turns to mush.

Reply:
We can maybe do a type parameter and some Aux magic and drop the Vault dependency?  (Vault would still be ideal for a pure Scala implementation, but I'm not going there anytime soon...)  But then I'm not sure how we'd create and reference keys in the core.